CardScan/Classes/API.swift

```diff
@@ -135,6 +135,61 @@ public struct Api {
         apiCall(endpoint: endpoint, parameters: apiParameters, completion: completion)
     }
     
+    static public func getModelDownloadConfig(endpoint: String, parameters: [String: Any], completion: @escaping ApiCompletion) {
+        if baseUrl == nil || apiKey == nil {
+            DispatchQueue.main.async { completion(nil, apiUrlNotSet) }
+            return
+        }
+        
+        modelDownloadConfig(endpoint: endpoint, parameters: parameters, completion: completion)
+    }
+    
+    private static func modelDownloadConfig(endpoint: String, parameters: [String: Any], completion: @escaping ApiCompletion) {
+        guard let baseUrl = self.baseUrl else {
+            DispatchQueue.main.async { completion(nil, apiUrlNotSet) }
+            return
+        }
+
+        guard let url = URL(string: baseUrl + endpoint) else {
```

Comment:
Can you add a function that is something like `urlWithQueryParameters(baseUrl, endpoint, parameters) -> URL?` so that we can unit test it? I saw from SO they suggest escaping `+` manually, but this is the sort of thing that we'd want to have a unit test for so putting this bit of code into its own function will make testing it easy